census/src/main/scala/grasshopper/census/search/AddressInterpolator.scala

```diff
@@ -8,9 +8,37 @@ import grasshopper.census.model.AddressRange
 object AddressInterpolator {
 
   def calculateAddressRange(f: Feature, a: Int): AddressRange = {
-    val addressIsEven = (a % 2 == 0)
-    val rightRangeIsEven = (f.values.getOrElse("RFROMHN", 0).toString.toInt % 2 == 0)
-    val leftRangeIsEven = (f.values.getOrElse("LFROMHN", 0).toString.toInt % 2 == 0)
+    val addressIsEven = a % 2 == 0
+    val rFromEven = f.values.getOrElse("RFROMHN", 0)
+    val rightFromRange = rFromEven match {
+      case "" => 0
+      case _ => rFromEven.toString.toInt
+    }
+    val rightFromIsEven = rFromEven != "" && rightFromRange % 2 == 0
+
+    val rToEven = f.values.getOrElse("RTOHN", 0)
+    val rightToRange = rFromEven match {
+      case "" => 0
+      case _ => rToEven.toString.toInt
+    }
+    val rightToIsEven = rToEven != "" && rightToRange % 2 == 0
+
+    val rightRangeIsEven = rightFromIsEven || rightToIsEven
+
+    val lFromEven = f.values.getOrElse("LFROMHN", 0)
+    val leftFromRange = lFromEven match {
+      case "" => 0
+      case _ => lFromEven.toString.toInt
+    }
+    val leftFromIsEven = lFromEven != "" && leftFromRange % 2 == 0
+
+    val ltoEven = f.values.getOrElse("LTOHN", 0)
+    val leftToRange = ltoEven match {
+      case "" => 0
+      case _ => ltoEven.toString.toInt
+    }
+    val leftToIsEven = ltoEven != "" && leftToRange % 2 == 0
+    val leftRangeIsEven = leftFromIsEven || leftToIsEven
```

Comment:
Looks like near-identical code repeated 4x.  Seems like a separate function is in order.